src/swap/components/SwapSettingsSlippageLayoutBottomSheet.tsx

```diff
@@ -0,0 +1,62 @@
+import { Children, cloneElement, isValidElement, useMemo } from 'react';
+import { findComponent } from '../../internal/utils/findComponent';
+import { cn } from '../../styles/theme';
+import type { SwapSettingsSlippageLayoutReact } from '../types';
+import { SwapSettingsSlippageDescription } from './SwapSettingsSlippageDescription';
+import { SwapSettingsSlippageInput } from './SwapSettingsSlippageInput';
+import { SwapSettingsSlippageTitle } from './SwapSettingsSlippageTitle';
+import { SwapSettingsSlippageToggle } from './SwapSettingsSlippageToggle';
+
+export function SwapSettingsSlippageLayoutBottomSheet({
+  children,
+  className,
+  customSlippageEnabled,
+  onToggleCustomSlippage,
+}: SwapSettingsSlippageLayoutReact) {
+  const { title, description, toggle, input } = useMemo(() => {
+    const childrenArray = Children.toArray(children);
+    return {
+      title: childrenArray.find(findComponent(SwapSettingsSlippageTitle)),
+      description: childrenArray.find(
+        findComponent(SwapSettingsSlippageDescription),
+      ),
+      toggle: childrenArray.find(findComponent(SwapSettingsSlippageToggle)),
+      input: childrenArray.find(findComponent(SwapSettingsSlippageInput)),
+    };
+  }, [children]);
+
+  const enhancedToggle = useMemo(() => {
+    if (isValidElement(toggle)) {
+      return cloneElement(toggle, {
+        customSlippageEnabled,
```

Comment:
This all `slippageEnable` story it's very confusing. I still don't understand why are we making this mess in passing this around.

If the component is there, than we want custom, if it's not there we don't want custom. End of the story. :)

Reply:
Ok I will clean this up

Reply:
Did a major cleanup. I combined the Input and Toggle sub-components. Was able to remove a ton of logic including customSlippage